{"version":3,"sources":["../src/extension.js"],"names":["configs","require","apiRoutes","express","middleware","auth","process","env","NODE_ENV","NODE_TLS_REJECT_UNAUTHORIZED","module","exports","extendApp","context","app","use","json","appConfig","handleConfigs","handleAuth","on","error","promise","console","detail","requestData","request","_currentUrl","e"],"mappings":";;AAAA,IAAMA,UAAUC,oBAAhB;AACA,IAAMC,YAAYD,gBAAlB;AACA,IAAIE,UAAUF,QAAQ,SAAR,CAAd;AACA,IAAIG,aAAaH,uBAAjB;AACA,IAAMI,OAAOJ,iBAAb;AACA,IAAIK,QAAQC,GAAR,CAAYC,QAAZ,KAAyB,aAA7B,EAA4C;AAC1CF,UAAQC,GAAR,CAAYE,4BAAZ,GAA2C,CAA3C;AACD;AACDC,OAAOC,OAAP,CAAeC,SAAf,GAA2B,UAAUC,OAAV,EAAmB;AAC5CA,UAAQC,GAAR,CAAYC,GAAZ,CAAgBZ,QAAQa,IAAR,EAAhB;AACAH,UAAQC,GAAR,CAAYC,GAAZ,CAAgBX,WAAWJ,QAAQiB,SAAnB,CAAhB;AACAjB,UAAQkB,aAAR,CAAsBL,OAAtB;AACAR,OAAKc,UAAL,CAAgBN,OAAhB;AACAA,UAAQC,GAAR,CAAYC,GAAZ,CAAgB,MAAhB,EAAwBb,SAAxB;AACD,CAND;AAOAI,QAAQc,EAAR,CAAW,oBAAX,EAAiC,UAACC,KAAD,EAAQC,OAAR,EAAoB;AACnD,MAAI;AACFC,YAAQF,KAAR,CAAc,+BAAd,EAA+C;AAC7CG,cAAQ;AACNC,qBAAaJ,SAASA,MAAMK,OAAf,IAA0BL,MAAMK,OAAN,CAAcC,WAD/C;AAENN,oBAFM;AAGNC;AAHM;AADqC,KAA/C;AAOD,GARD,CAQE,OAAOM,CAAP,EAAU;AACVL,YAAQF,KAAR,CAAc,+BAAd,EAA+C,EAAEG,QAAQH,KAAV,EAAiBO,GAAGA,CAApB,EAA/C;AACD;AACF,CAZD","file":"extension.js","sourcesContent":["const configs = require('./configs')\nconst apiRoutes = require('./api')\nvar express = require('express')\nvar middleware = require('./middleware')\nconst auth = require('./auth')\nif (process.env.NODE_ENV === 'development') {\n  process.env.NODE_TLS_REJECT_UNAUTHORIZED = 0\n}\nmodule.exports.extendApp = function (context) {\n  context.app.use(express.json())\n  context.app.use(middleware(configs.appConfig))\n  configs.handleConfigs(context)\n  auth.handleAuth(context)\n  context.app.use('/api', apiRoutes)\n}\nprocess.on('unhandledRejection', (error, promise) => {\n  try {\n    console.error('Unhandled rejection in Server', {\n      detail: {\n        requestData: error && error.request && error.request._currentUrl,\n        error,\n        promise\n      }\n    })\n  } catch (e) {\n    console.error('Unhandled rejection in Server', { detail: error, e: e })\n  }\n})\n"]}